modules/standard/HDFS.chpl

```diff
@@ -18,19 +18,315 @@
+
+   * Now all we need to do is format the namenode and start things up:
+
+     .. code-block:: sh
+
+       hadoop namenode -format
+       start-all.sh  # (This will start hdfs and the tasktracker/jobtracker)
+
+   * In general, hadoop has the same type of commands as bash,
+     usually in the form:
+
+     .. code-block:: sh
+
+         hadoop dfs -<command> <regular args to that command>
+
+   * At this point, you can compile and run Chapel programs using HDFS
+
+   * You can check the status of Hadoop via http, for example on a local
+     host (e.g., for ea above), using:
+
+       *  http://localhost:50070/
+       *  http://localhost:50030/
+
+     For cluster mode (3b), you'll use the name of the master host in the
+     URL and its port (see the web for details).
+
+5. Shut things down:
+
+   .. code-block:: sh
+
+     stop-all.sh   # (This will stop hdfs and mapreduce)
+
+6. Set up Chapel to run in distributed mode:
+
+   * You'll need to set up your Chapel environment to target multiple
+     locales in the standard way (see README.multilocale and the
+     "Settings to run Chapel on multiple nodes" section of the
+     .bashrc below).
+
+   * After this you should be able to run Chapel code with HDFS over
+     a cluster of computers the same way as you normally would.
+    
+
+Here is a sample .bashrc for using Hadoop within Chapel:
+
+
+.. code-block:: sh
+
+  #
+  # For Hadoop
+  #
+  export HADOOP_INSTALL=<Place where you have Hadoop installed>
+  export HADOOP_HOME=$HADOOP_INSTALL
+  export HADOOP_VERSION=<Your Hadoop version number>
+  #
+  # Note that the following environment variables might contain more paths than
+  # those listed below if you have more than one IO system enabled. These are all
+  # that you will need in order to use HDFS (only)
+  #
+  export CHPL_AUXIO_INCLUDE="-I$JAVA_INSTALL/include -I$JAVA_INSTALL/include/linux  -I$HADOOP_INSTALL/src/c++/libhdfs"
+  export CHPL_AUXIO_LIBS="-L$JAVA_INSTALL/jre/lib/amd64/server -L$HADOOP_INSTALL/c++/Linux-amd64-64/lib"
+  
+  #
+  # So we can run things such as start-all.sh etc. from anywhere and
+  # don't need to be in $HADOOP_INSTALL
+  #
+  export PATH=$PATH:$HADOOP_INSTALL/bin
+  
+  #
+  # Point to the JDK installation
+  #
+  export JAVA_INSTALL=<Place where you have the jdk installed>
+  
+  #
+  # Add Hadoop directories to the Java class path
+  #
+  export CLASSPATH=$CLASSPATH:$HADOOP_HOME/""*:$HADOOP_HOME/lib/""*:$HADOOP_HOME/conf/""*:$(hadoop classpath):
+  
+  #
+  # So we don't have to "install" these things
+  #
+  export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$HADOOP_HOME/c++/Linux-amd64-64/lib:$HADOOP_HOME/src/c++/libhdfs:$JAVA_INSTALL/jre/lib/amd64/server:$JAVA_INSTALL:$HADOOP_HOME/lib:$JAVA_INSTALL/jre/lib/amd64:$CLASSPATH
+  
+  #
+  # Settings to run Chapel on multiple nodes
+  #
+  export GASNET_SPAWNFN=S
+  export SSH_SERVERS=<the names of the computers in your cluster>
+  export SSH_CMD=ssh
+  export SSH_OPTIONS=-x
+  export GASNET_ROUTE_OUTPUT=0
+
```

Comment:
Similar to my comment in #1661, you may want to add a section here for "functions" or something, so that it's easy to get to and find.